Editor: sync latest packages to trunk

ellatrix opened the pull request:
See https://make.wordpress.org/core/handbook/about/release-cycle/block-editor-release-process-for-major-releases/#package-updates-and-core-patches

```
npm run sync-gutenberg-packages -- --dist-tag=latest
```

<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/64303

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

ellatrix committed:
Editor: sync latest packages to trunk

ellatrix commented:
One failure seems due to https://github.com/WordPress/gutenberg/commit/efb328b100eed811d0417a662680c0d938c14247, maybe the snapshots need to be updated in trunk.

sirreal committed:
Backport changes from Gutenberg PR 72485.

The PR changed the exports from the @wordpress/interactivity package.
It removed the `debug` WordPress export module completely.

This requires some changes to the special way the debug script module
was being registered.

See https://github.com/WordPress/gutenberg/pull/72485.

sirreal committed:
Update tests for removed Interactivity debug module.

The debug module was removed, only the main interactivity module is
exposed.

sirreal commented:
I've pushed two commits to backport necessary changes from https://github.com/WordPress/gutenberg/pull/72485.

These were required because the `@wordpress/interactivity` module changed
`wpScriptModuleExports` in its package.json and removed an exposed module.

There was some special handling in place for its debug module that had to be removed.

@youknowriad it would be great if you could take a look at the backports I've added to this PR.

sirreal commented on src/wp-includes/script-modules.php:
```diff
@@ -159,30 +159,11 @@ function wp_default_script_modules() {
+		 *   - interactivity/index.min.js or interactivity/index.js => @wordpress/interactivity
+		 *   - block-library/query/view.min.js or block-library/query/view.js => @wordpress/block-library/query/view
 		 */
 		$script_module_id = '@wordpress/' . preg_replace( '~(?:/index)?(?:\.min)?\.js$~D', '', $file_name, 1 );
```
https://github.com/WordPress/gutenberg/pull/72485 changed this line which I did _not_ copy, as far as I can tell the change is not significant: https://github.com/WordPress/gutenberg/pull/72485/files#r2560447901

Maybe it should be updated just for the sake of consistency:

```suggestion
		$script_module_id = '@wordpress/' . preg_replace( '~(?:/index)?\.(min\.)?js$~D', '', $file_name, 1 );
```

sirreal commented:
There's a remaining failure unrelated to Interactivity or Modules:

```
There was 1 failure:

1) Tests_Blocks_Render::test_do_block_output with data set #6 ('core__categories.html', 'core__categories.server.html')
File '/var/www/tests/phpunit/includes/../data/blocks/fixtures/core__categories.html' does not match expected value
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<ul class="wp-block-categories-list wp-block-categories"><li class="cat-item-none">No categories</li></ul>'
+'<ul class="wp-block-categories-list wp-block-categories-taxonomy-category wp-block-categories"><li class="cat-item-none">No categories</li></ul>'
```

sirreal commented:
The categories class change appears to be from https://github.com/WordPress/gutenberg/pull/72662.

The categories block `index.php` changed across these packages:

```diff
--- 1/index.php
+++ 2/index.php	
@@ -79,7 +79,7 @@ function render_block_core_categories( $attributes, $content, $block ) {
 		}
 	}
 
-	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => "wp-block-categories-{$type}" ) );
+	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => "wp-block-categories-{$type} wp-block-categories-taxonomy-{$attributes['taxonomy']}" ) );
 
 	return sprintf(
 		$wrapper_markup,
```

I'll push a commit to update the failing test assertion. FYI @ntsekouras.

sirreal committed:
Update categories server HTML fixture from GB 72662

Gutenberg PR 72662 updated the rendering of the categories block adding
this class.

See https://github.com/WordPress/gutenberg/pull/72662.

ntsekouras commented:
>I'll push a commit to update the failing test assertion. FYI @ntsekouras.

Thanks. That sounds fine!

ellatrix commented:
Currently the editors are not loading:

> Error: You tried to opt-in to unstable APIs as module "@wordpress/dataviews" which is already registered. This feature is only for JavaScript modules shipped with WordPress core. Please do not use it in plugins and themes as the unstable APIs will be removed without a warning. If you ignore this error and depend on unstable features, your product will inevitably break on one of the next WordPress releases.

See https://github.com/WordPress/gutenberg/pull/73554

ellatrix committed:
Update with latest packages
